modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Button.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import React from 'react';
+
+export default function Button({
+	componentId: _componentId,
+	cssClass,
+	icon,
+	label,
+	locale: _locale,
+	portletId: _portletId,
+	portletNamespace: _portletNamespace,
+	...otherProps
+}) {
+	return icon ? (
+		<ClayButton className={cssClass} {...otherProps}>
+			<span className="inline-item inline-item-before">
```

Comment:
We've seen this before where we add the `inline-item-before` and there's no `label` and then we end up with a bigger button than expected because of that unconditional padding.

Reply:
Here's how we managed it https://github.com/liferay/liferay-portal/blob/3c6bbac29b64769b665e37798c7f51d4e45b6ff0/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/DropdownMenu.js#L61

Reply:
To be honest, I hadn't thought about that although I did have in mind that this might need to be changed. 
I'm going to try with what @carloslancha pointed out. Thanks